File: backend/GeoQuiz_backend/GeoQuiz_backend/Application/DTOs/User/LeaderboardDto.cs
namespace GeoQuiz_backend.Application.DTOs.User
{
    public class LeaderboardDto
    {
        public List<LeaderboardEntry> LeaderboardEntries { get; set; } = new List<LeaderboardEntry>();
        public int yourRank { get; set; }
        public int yourScore { get; set; }
    }
    public class LeaderboardEntry
    {
        public Guid playerId { get; set; }
        public string playerName { get; set; } = null!;
        public int rank { get; set; }
        public int level { get; set; }
        public int totalScore { get; set; }
    }
}

File: frontend/app/src/main/java/com/example/geoquiz_frontend/data/remote/dtos/profile/LeaderboardDto.java
package com.example.geoquiz_frontend.data.remote.dtos.profile;

import com.example.geoquiz_frontend.presentation.ui.leaderboard.LeaderboardEntry;

import java.util.List;

public class LeaderboardDto {
    private List<LeaderboardEntry> leaderboardEntries;
    private int yourRank;
    private int yourScore;
    public List<LeaderboardEntry> getLeaderboardEntries() { return leaderboardEntries;}
    public int getYourRank() { return yourRank; }
    public int getYourScore() { return yourScore; }
}
